zerver/decorator.py

```diff
@@ -527,7 +527,8 @@ def authenticate_log_and_execute_json(request, view_func, *args, **kwargs):
     if user_profile.is_incoming_webhook:
         raise JsonableError(_("Webhook bots can only access webhooks"))
 
-    process_client(request, user_profile, is_browser_view=True)
+    process_client(request, user_profile, is_browser_view=True,
+                   query=view_func.__name__)
```

Comment:
If you read the code, I'm pretty sure that the old code path set this to `request.META['PATH_INFO']`, not the name of the view func... I am not sure whether that will cause problems, but it seems wholly possible it would break the `zerver_useractivity` code path in `analytics/views.py` that check for e.g. the pointer position in certain formats.

(for background, Zulip originally only supported the `PATH_INFO` version before we added the REST API).

Reply:
I think the fix for `analytics/views.py` is that these blocks:
```
                            '/json/send_message',                                                                                                              
                            'send_message_backend',                                                                                                            
                            '/api/v1/send_message',                                                                                                            
                            '/json/update_pointer',                                                                                                            
                            '/json/users/me/pointer'                                                                                                           
```
will just need the new value for moving the pointer, presumably `update_pointer_backend`, added.

Reply:
Huh. I'm looking through `urls.py`; it looks like just one of those URLs is mapped to anything today: `/json/users/me/pointer`. (I imagine the rest are from old versions.)

But then that one uses `rest_dispatch`:
```
    url(r'^users/me/pointer$', rest_dispatch,
        {'GET': 'zerver.views.pointer.get_pointer_backend',
         'PUT': 'zerver.views.pointer.update_pointer_backend',
         'POST': 'zerver.views.pointer.update_pointer_backend'}),
```
so those should be logged by the names `get_pointer_backend` or `update_pointer_backend`. And neither of those is on the list!

So it *looks* to me like we should be adding `'update_pointer_backend'` to this list. But that should be true already in master.

Reply:
Well, I don't really believe in leaving broken things we've noticed broken, so I went ahead and added 732a5bae84aa9ac98990363c556397a414f6893a to fix this.

Reply:
Great -- but if that's right, didn't that mean this feature was totally broken? Was it in fact totally broken?

Reply:
What it meant is that we were just displaying the last time the user sent a message, not the last time they sent a message or read something.  So, broken, but not in a way that would be obvious on inspection.

Reply:
Ah, cool, that makes sense.